### PR TITLE
build: Use WARN_CFLAGS variable

### PR DESCRIPTION
**Pull request by rbuj**
Test:
```
$ ./autogen.sh  --prefix=/usr --enable-compile-warnings=maximum && make V=1
```

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -10,6 +10,8 @@ m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 AC_CONFIG_HEADERS(config.h)
 AC_CONFIG_MACRO_DIR([m4])
 
+MATE_COMPILE_WARNINGS([yes])
```
Comment (raveit65) [inline configure.ac:13]: `yes` is always set as default by mate-common.
`[enable_compile_warnings="m4_default([$1],[yes])"])`
https://github.com/mate-desktop/mate-common/blob/master/macros/mate-compiler-flags.m4#L37
But OK, not an issue :smile:

Comment (rbuj) [reply configure.ac:13]: I'm going to change configure.ac files to use mate-common default vaule.
```
$ grep -R MATE_COMPILE_WARNINGS --include configure.ac
mate-polkit/configure.ac:MATE_COMPILE_WARNINGS([maximum])
mate-session-manager/configure.ac:MATE_COMPILE_WARNINGS(maximum)
libmateweather/configure.ac:MATE_COMPILE_WARNINGS
mate-menus/configure.ac:MATE_COMPILE_WARNINGS(yes)
mate-calc/configure.ac:MATE_COMPILE_WARNINGS
mate-sensors-applet/configure.ac:MATE_COMPILE_WARNINGS(maximum)
engrampa/configure.ac:MATE_COMPILE_WARNINGS
mate-terminal/configure.ac:MATE_COMPILE_WARNINGS([yes])
libmatemixer/configure.ac:MATE_COMPILE_WARNINGS([yes])
mate-control-center/configure.ac:MATE_COMPILE_WARNINGS
mate-notification-daemon/configure.ac:MATE_COMPILE_WARNINGS([yes])
mate-applets/configure.ac:MATE_COMPILE_WARNINGS
marco/configure.ac:MATE_COMPILE_WARNINGS([yes])
mate-screensaver/configure.ac:MATE_COMPILE_WARNINGS([yes])
eom/configure.ac:MATE_COMPILE_WARNINGS
caja/configure.ac:MATE_COMPILE_WARNINGS
atril/configure.ac:MATE_COMPILE_WARNINGS([yes])
mate-system-monitor/configure.ac:MATE_COMPILE_WARNINGS([yes])
mate-utils/configure.ac:MATE_COMPILE_WARNINGS([yes])
mate-power-manager/configure.ac:MATE_COMPILE_WARNINGS([yes])
pluma/configure.ac:MATE_COMPILE_WARNINGS(yes)
libmatekbd/configure.ac:MATE_COMPILE_WARNINGS
mate-settings-daemon/configure.ac:MATE_COMPILE_WARNINGS
mate-media/configure.ac:MATE_COMPILE_WARNINGS([yes])
mate-desktop/configure.ac:MATE_COMPILE_WARNINGS
mate-panel/configure.ac:MATE_COMPILE_WARNINGS(maximum)
```

Comment (raveit65) [reply configure.ac]: Yes, go ahead, but please commit without PR ;)

Comment (raveit65) [reply configure.ac]: Btw, i am working on mate-panel currently.

**Pull request merged**